### ISSUE TITLE
Context menu popup customisation

### ISSUE THREAD

**Issue by ability-app** [PDFTron/webviewer-ui#387]
It should be possible to customise context menu popups event the existing ones (TextPopup, AnnotationPopup).

Now they are hardcoded, can be easily solved with a configuration object like the custom header.

**Comment by ZhijieZhang** [comment]
Yes, we are aware of this issue. In the 6.0 beta version you can use APIs to add, update, delete buttons in these three popups. The API looks like:

```javascript
// add
instance.contextMenuPopup.add({
  type: 'actionButton',
  img: 'path/to/image',
  onClick: () => {...}
}, 'a dataElement of a button in this popup')

// update
instance.contextMenuPopup.update('a dataElement of a button in this popup', {
  img: '...',
  title: '...'
})

// delete
instance.contextMenuPopup.delete('a dataElement of a button in this popup');
```
Unfortunately this API looks different than the header API(setHeaderItems). The reason we didn't introduce a breaking changing in 6.0 to replace `setHeaderItems` with `instance.header` for API consistency is because the header API may change again in 7.0 due to a UI redesign and we don't want to change this API that often.

As for the 6.0 beta, we plan to release this in a few weeks. Let me know if you have any questions! Thanks for bringing this up!

**Comment by ZhijieZhang** [comment]
Hi,

We just released the 6.0-beta, which can be downloaded from https://www.pdftron.com/downloads/WebViewer-6.0.0-beta-1.zip. 
We also have a migration guide here https://www.pdftron.com/documentation/web/get-started/migrating-to-v6. 
You can view the customization you can do for the contextMenuPopup here https://www.pdftron.com/api/web/6.0/WebViewer.Popup.html

Can you give it a try and let us know how it goes? Thanks!

**Comment by ZhijieZhang** [comment]
I will close this issue for now as it should have been addressed in the beta release. Please feel free to reopen it if you found any issues. Thanks! :)
